scipy/interpolate/_bsplines.py

```diff
@@ -445,34 +476,87 @@ def integrate(self, a, b, extrapolate=None):
         if extrapolate is None:
             extrapolate = self.extrapolate
 
-        if not extrapolate:
-            # shrink the integration interval, if needed
+        # Prepare self.t and self.c.
+        self._ensure_c_contiguous()
+
+        # Swap integration bounds if needed.
+        sign = 1
+        if b < a:
+            a, b = b, a
+            sign = -1
+        n = self.t.size - self.k - 1
+
+        if not extrapolate and (extrapolate !="periodic"):
+            # Shrink the integration interval, if needed.
             a = max(a, self.t[self.k])
-            b = min(b, self.t[-self.k - 1])
+            b = min(b, self.t[n])
 
             if self.c.ndim == 1:
-                # fast path: use FITPACK's routine (cf _fitpack_impl.splint)
+                # Fast path: use FITPACK's routine
+                # (cf _fitpack_impl.splint).
                 t, c, k = self.tck
-                aint, wrk = _dierckx._splint(t, c, k, a, b)
-                return aint
+                integral, wrk = _dierckx._splint(t, c, k, a, b)
```

Comment:
Obviously we want to return `sign * integral` here. It was the point to moving it here.